packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -176,21 +176,17 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
       ? `'modulepreload'`
       : `(${detectScriptRel.toString()})()`
 
-  // There are three different cases for the preload list format in __vitePreload
+  // There are two different cases for the preload list format in __vitePreload
   //
   // __vitePreload(() => import(asyncChunk), [ ...deps... ])
   //
   // This is maintained to keep backwards compatibility as some users developed plugins
   // using regex over this list to workaround the fact that module preload wasn't
   // configurable.
-  const assetsURL = renderBuiltUrl
-    ? // If `experimental.renderBuiltUrl` is used, the dependencies are already resolved.
-      // To avoid the need for `new URL(dep, import.meta.url)`, a helper `__vitePreloadRelativeDep` is
-      // used to resolve from relative paths which can be minimized.
-      `function(dep, importerUrl) { return dep[0] === '.' ? new URL(dep, importerUrl).href : dep }`
-    : optimizeModulePreloadRelativePaths
-      ? // If there isn't custom resolvers affecting the deps list, deps in the list are relative
-        // to the current chunk and are resolved to absolute URL by the __vitePreload helper itself.
+  const assetsURL =
+    renderBuiltUrl || optimizeModulePreloadRelativePaths
```

Comment:
Unrelated, but I realized that all the `optimizeModulePreloadRelativePaths` usage could be replaced with `isRelativeBase` instead, which makes it more readable. I don't mind if we update here along or in a separate PR though.

Reply:
Ah, didn't notice. Updated 👍